democracy/views/comment.py

```diff
@@ -137,22 +148,27 @@ def perform_update(self, serializer):
 
     @detail_route(methods=['post'])
     def vote(self, request, **kwargs):
-        # Return 403 if user is not authenticated
-        if not request.user.is_authenticated():
-            return response.Response({'status': 'Forbidden'}, status=status.HTTP_403_FORBIDDEN)
-
+        resp = self._check_may_vote(request)
+        if resp:
+            return resp
         comment = self.get_object()
 
-        # Check if user voted already. If yes, return 400.
-        if comment.__class__.objects.filter(id=comment.id, voters=request.user).exists():
-            return response.Response({'status': 'Already voted'}, status=status.HTTP_304_NOT_MODIFIED)
-
-        # add voter
-        comment.voters.add(request.user)
-        # update number of votes
-        comment.recache_n_votes()
-        # return success
-        return response.Response({'status': 'Vote has been added'}, status=status.HTTP_201_CREATED)
+        if not request.user.is_authenticated():
+            # If the check went through, anonymous voting is allowed
+            comment.n_unregistered_votes += 1
+            comment.recache_n_votes()
+            return response.Response({'status': 'Vote has been counted'}, status=status.HTTP_200_OK)
+        else:
```

Comment:
Unnecessary `else:`; the previous branch has a `return`. (Removing the `else:` will also have the pleasant side effect of shortening the diff, as the rest of the function remains untouched.)